### PR TITLE
Remove jQuery `.attr` from the code line range selection

### PR DESCRIPTION
**Pull request by yardenshoham**
- Switched from jQuery `attr` to plain javascript `getAttribute` and `setAttribute`
- Tested the code line range selection and it works as before

### CODE REVIEW

File: web_src/js/features/repo-code.js [hunk 1]
```diff
@@ -28,40 +28,40 @@ function selectRange($linesEls, $selectionEndEl, $selectionStartEls) {
...
-  const $viewGitBlame = $('a.view_git_blame');
+  const refInNewIssue = document.querySelector('a.ref-in-new-issue');
+  const copyPermalink = document.querySelector('a.copy-line-permalink');
+  const viewGitBlame = document.querySelector('a.view_git_blame');
 
   const updateIssueHref = function (anchor) {
-    if (!$refInNewIssue.length) {
+    if (!refInNewIssue) {
       return;
     }
```
Comment (silverwind) [inline web_src/js/features/repo-code.js]:
```suggestion
    if (!refInNewIssue) return;
```

File: web_src/js/features/repo-code.js [hunk 2]
```diff
@@ -28,40 +28,40 @@ function selectRange($linesEls, $selectionEndEl, $selectionStartEls) {
...
       href = `${href}#${anchor}`;
     }
-    $viewGitBlame.attr('href', href);
+    viewGitBlame.setAttribute('href', href);
   };
 
-  const updateCopyPermalinkUrl = function(anchor) {
-    if (!$copyPermalink.length) return;
-    let link = $copyPermalink.attr('data-url');
+  const updateCopyPermalinkUrl = function (anchor) {
```
Comment (silverwind) [inline web_src/js/features/repo-code.js:53]: I guess there must be a lint rule for this space 😉

**Review by silverwind — APPROVED** [review]

File: web_src/js/features/repo-code.js (hunk 2 above)
Comment (silverwind) [reply web_src/js/features/repo-code.js:53]: Thought I'm torn whether to enable this rule. My personal config has

```yaml
"@stylistic/js/space-before-function-paren": [2, {anonymous: never, named: never, asyncArrow: always}]
```

The dominant style in gitea is

```yaml
"@stylistic/js/space-before-function-paren": [2, {anonymous: always, named: never, asyncArrow: always}]
```

So I guess I keep it optional for now. With the removal of jQuery, the amount of anonymous non-arrow functions should become zero.

Comment (silverwind) [reply web_src/js/features/repo-code.js:53]: https://github.com/go-gitea/gitea/pull/30078 will enable the rule but ignore anonymous.

**Review by delvh — APPROVED** [review]

**Pull request merged**